scipy/signal/signaltools.py

```diff
@@ -2843,24 +2848,49 @@ def decimate(x, q, n=None, ftype='iir', axis=-1):
     --------
     resample
     resample_poly
+
+    Notes
+    -----
+    The ``zero_phase`` keyword was added in 0.18.0.
+    The possibility to use instances of ``lti`` as ``ftype`` was added in
+    0.18.0.
     """
 
     if not isinstance(q, int):
         raise TypeError("q must be an integer")
 
-    if n is None:
-        if ftype == 'fir':
+    if ftype == 'fir':
+        if n is None:
             n = 30
-        else:
-            n = 8
+        system = lti(firwin(n + 1, 1. / q, window='hamming'), 1.)
 
-    if ftype == 'fir':
-        b = firwin(n + 1, 1. / q, window='hamming')
-        a = 1.
+    elif ftype == 'iir':
+        if n is None:
+            n = 8
+        system = lti(*cheby1(n, 0.05, 0.8 / q))
+    elif isinstance(ftype, lti):
+        system = ftype
     else:
-        b, a = cheby1(n, 0.05, 0.8 / q)
-
-    y = lfilter(b, a, x, axis=axis)
+        raise ValueError('invalid ftype')
+
+    if zero_phase is None:
+        warnings.warn("""
+                      Note: Decimate's zero_phase keyword argument will default
+                      to True in a future release.  Until then, decimate
+                      defaults to one-way filtering for backwards
+                      compatibility. Ideally, always set this argument
+                      explicitly.""", FutureWarning)
+        zero_phase = False
+
+    if zero_phase is True and ftype == 'fir':
```

Comment:
I slightly prefer `if zero_phase` to `if zero_phase is True` because it allows for reasonable/expected behavior for e.g. `zero_phase=1` and `zero_phase=0`.